rotkehlchen/globaldb/handler.py

```diff
@@ -1011,6 +1011,108 @@ def add_historical_prices(entries: List['HistoricalPrice']) -> None:
 
         connection.commit()
 
+    @staticmethod
+    def add_single_historical_price(entry: HistoricalPrice) -> None:
+        """Adds the given historical price entries in the DB"""
+        connection = GlobalDBHandler()._conn
+        cursor = connection.cursor()
+        try:
+            serialized = entry.serialize_for_db()
+            cursor.execute(
+                """INSERT OR IGNORE INTO price_history(
+                from_asset, to_asset, source_type, timestamp, price
+                ) VALUES (?, ?, ?, ?, ?)
+                """,
+                serialized,
+            )
+        except sqlite3.IntegrityError as e:
+            connection.rollback()
+            log.error(
+                f'Failed to add single historical price. {str(e)}. ',
+            )
```

Comment:
You should return here, before the DB commit

```suggestion
            )
            return  #  when you change it to return success or failure this should be false
```